superset/assets/javascripts/SqlLab/actions.js

```diff
@@ -382,3 +386,38 @@ export function popSavedQuery(saveQueryId) {
     });
   };
 }
+
+export function createDatasourceStarted() {
+  return { type: CREATE_DATASOURCE_STARTED };
+}
+export function createDatasourceSuccess(response) {
+  const data = JSON.parse(response);
+  const datasource = `${data.table_id}__table`;
+  return { type: CREATE_DATASOURCE_SUCCESS, datasource };
+}
+export function createDatasourceFailed(err) {
+  return { type: CREATE_DATASOURCE_FAILED, err };
+}
+
+export function createDatasource(vizOptions, context) {
+  return (dispatch) => {
+    dispatch(createDatasourceStarted());
+
+    return $.ajax({
+      type: 'POST',
+      url: '/superset/sqllab_viz/',
+      async: false,
+      data: {
+        data: JSON.stringify(vizOptions),
+      },
+      context,
```

Comment:
i don't think we need to set context here, since we are not using `this` in the `success` or `error` callbacks. 
https://stackoverflow.com/questions/5097191/ajax-context-option#5097214